kernel/contracts/KernelRegistry.sol

```diff
@@ -0,0 +1,25 @@
+pragma solidity ^0.4.18;
+
+import "./KernelInstance.sol";
+import "zeppelin-solidity/contracts/ownership/Ownable.sol";
+
+contract KernelRegistry is Ownable {
+  event NewInstance(KernelInstance indexed instance);
+
+  mapping(bytes32 => address) private instances;
+
+  function KernelRegistry() public {}
+
+  function getInstance(string name, string version) public view returns(KernelInstance) {
+    bytes32 hash = keccak256(name, version);
+    return KernelInstance(instances[hash]);
+  }
+
+  function addInstance(KernelInstance _instance) onlyOwner public {
+    bytes32 hash = _instance.getHash();
```

Comment:
Better calculate the hash in this contract, in case the kernel instance does something weird on getHash calculation

Reply:
We actually resorted to using (`bytes32`) hashes instead of name/version because of Solidity's limitation in returning dynamically sized arrays (such as strings) from one contract to another.. So we cannot here call `_instance.name()` or `_instance.version()` to compute the hash locally?

Any other strategies around this?

Reply:
Good point. Let's stick with this for now.